gender_analysis/analysis/dependency_parsing.py

```diff
@@ -92,28 +87,43 @@ def _get_parser_download_if_not_present():
     return parser
 
 
-def generate_dependency_tree(document, pickle_filepath=None):
+def generate_dependency_tree(document, genders=None, pickle_filepath=None):
     """
-    This function returns the dependency tree for a given document.
+    This function returns the dependency tree for a given document. This can optionally be reduced
+    such that it will only analyze sentences that involve specified genders' subject/object
+    pronouns.
 
     :param document: Document we are interested in
+    :param genders: a collection of genders that will be used to filter out sentences that do not \
+        involve the provided genders. If set to `None`, all sentences are parsed (default).
     :param pickle_filepath: filepath to store pickled dependency tree, will not write a file if None
     :return: dependency tree, represented as a nested list
 
     """
 
     parser = _get_parser_download_if_not_present()
     sentences = sent_tokenize(document.text.lower().replace("\n", " "))
-    he_she_sentences = []
-    for sentence in sentences:
-        add_sentence = False
-        words = [word for word in word_tokenize(sentence)]
-        for word in words:
-            if word == "he" or word == "she" or word == "him" or word == "her":
-                add_sentence = True
-        if add_sentence:
-            he_she_sentences.append(sentence)
-    sentences = he_she_sentences
+
+    # filter out sentences that are not relevant
+    if genders is not None:
```

Comment:
Note for later refactor: we may want to factor this out into its own function and call as a preprocessing step elsewhere. I'm guessing it was only here because dependency parsing is the most expensive thing this system does.